modin/core/storage_formats/pandas/aggregations.py

```diff
@@ -0,0 +1,590 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Contains implementations for aggregation functions."""
+
+import pandas
+from pandas.core.dtypes.common import is_numeric_dtype
+import numpy as np
+from typing import TYPE_CHECKING, Tuple, Callable
+from enum import Enum
+
+if TYPE_CHECKING:
+    from .query_compiler import PandasQueryCompiler
+
+from modin.utils import MODIN_UNNAMED_SERIES_LABEL
+
+
+class CorrCovBuilder:
+    """Responsible for building pandas query compiler's methods computing correlation and covariance matrices."""
+
+    class Method(Enum):
+        """Enum specifying what method to use (either CORR for correlation or COV for covariance)."""
+
+        CORR = 1
+        COV = 2
+
+    @classmethod
+    def build_corr_method(
+        cls,
+    ) -> Callable[["PandasQueryCompiler", str, int], "PandasQueryCompiler"]:
+        """
+        Build a query compiler method computing the correlation matrix.
+
+        Returns
+        -------
+        callable(qc: PandasQueryCompiler, method: str, min_periods: int) -> PandasQueryCompiler
+            A callable matching the ``BaseQueryCompiler.corr`` signature and computing the correlation matrix.
+        """
+
+        def corr_method(
+            qc: "PandasQueryCompiler", method: str, min_periods: int = 1
+        ) -> "PandasQueryCompiler":
+            if method != "pearson":
+                return super(type(qc), qc).corr(
+                    method=method, min_periods=min_periods, numeric_only=True
```

Comment:
that's because we only end up in the query compiler if `numeric_only=True` https://github.com/modin-project/modin/blob/cc69b3d78f7cd708d2c74ff5ccab8a50196e2d3c/modin/pandas/dataframe.py#L718-L736

Reply:
So we can just pass numeric_only as is through?